### DISCUSSION TITLE
Not able to fetch total_ram

### DISCUSSION THREAD

**Discussion by NCaur** [amidaware/tacticalrmm#1979] [General]
**Server Info (please complete the following information):**
 - OS: Windows  Pro
 - Browser:  chrome
 - RMM Version: 0.18.2

**Installation Method:**
  - [ ] Standard

**Agent Info (please complete the following information):**
Agents are having different versions and OS.

**Describe the bug**
I am trying to fetch the data using the API: **https://api.[CompanyDomain]/agents/-h 'X-API-KEY: [API KEY]'**

**To Reproduce**
curl https://api.[CompanyDomain]/agents/-h 'X-API-KEY: [API KEY]'
**Expected behavior**
It should display information like sitename, hostname, agentID,... including total_ram; but whenever I am trying to parse the value in total_ram; I am getting undefined; checked into the raw data, but nothing like total_ram is there.

Snippet from my curl response:

{"agent_id":"0***","alert_template":null,"hostname":"***","site_name":"***","client_name":"*****","monitoring_type":"workstation","description":"***","needs_reboot":false,"pending_actions_count":1,"status":"overdue","overdue_text_alert":false,"overdue_email_alert":false,"overdue_dashboard_alert":false,"last_seen":"2024-02-15T12:55:15.110723Z","boot_time":1708000484,"checks":{"total":0,"passing":0,"failing":0,"warning":0,"info":0,"has_failing_checks":false},"maintenance_mode":false,"logged_username":"**","italic":false,"block_policy_inheritance":false,"plat":"windows","goarch":"amd64","has_patches_pending":false,"version":"2.4.9","operating_system":"Windows 10 Pro, 64 bit v22H2 (build 19045.3324)","public_ip":"****","cpu_model":["***],"graphics":"****","local_ips":"****","make_model":"****","physical_disks":["I"],"custom_fields":[],"serial_number":"***DZ"}

**Comment by wh1te909** [comment]
this is not the right endpoint to get the `total_ram`. you can use one of the agent detail endpoints or use one of the beta api endpoints https://docs.tacticalrmm.com/functions/api/#beta-api
